bin/zowe-verify-authenticity.sh

```diff
@@ -0,0 +1,374 @@
+            outputPath=$dir/fingerprint
+            break
+        fi
+    done
+    if [[ ! -n "$outputPath" ]] # still failed to create a directory
+    then
+        echo Error: Cannot create default outputPath directory 
+        exit 1
+    fi
+else # create specified directory
+    mkdir -p $outputPath 1> /dev/null 2>/dev/null 
+    if [[ $? -ne 0 ]]
+    then
+        echo Error: Cannot create specified outputPath directory $outputPath 
+        exit 1
+    fi     
+fi
+outputPath=$(cd $outputPath;pwd)    # expand tilde and relative pathnames
+
+
+# is outputPath contained in runtime?
+echo $outputPath | grep ^$runtimePath  1> /dev/null 2> /dev/null
+if [[ $? -eq 0 ]]
+then
+    echo Error: outputPath $outputPath must not be in runtimePath $runtimePath
+    exit 1
+fi
+
+LOG_FILE=$outputPath/$SCRIPT.log
+touch $LOG_FILE
+if [[ $? -ne 0 ]]
+then
+    echo Error: Cannot write to $outputPath
+    exit 1
+fi
+
+echo Info: Logging to $LOG_FILE
+echo "<$SCRIPT.sh>"                 >  $LOG_FILE
+
+echo `date`                         >> $LOG_FILE
+echo "runtimePath = $runtimePath"   >> $LOG_FILE
+echo "hashPath    = $hashPath"      >> $LOG_FILE
+echo "refPath     = $refPath"       >> $LOG_FILE
+echo "outputPath  = $outputPath"    >> $LOG_FILE
+
+# - - - - - - - - - - - - - main logic
+
+# Verify runtime directory contents minimally
+for dir in bin components scripts manifest.json
+do
+    ls -l $runtimePath | grep " $dir$" 1> /dev/null 2>/dev/null
+    if [[ $? -ne 0 ]]
+    then
+        echo Error: runtimePath $runtimePath does not contain $dir | tee -a $LOG_FILE
+        exit 1
+    fi 
+done 
+
+# Reference hash file will be named like RefRuntimeHash-1.12.0.txt
+# manifest.json will contain a line like this near the top:
+#   "version": "1.11.0",
+# Extract Zowe version from manifest.json: 
+
+zoweVersion=`head $runtimePath/manifest.json | sed -n 's/ *"version" *: *"\(.*\)".*/\1/p'`
+if [[ -n $zoweVersion ]]
+then 
+    echo Info: zoweVersion = $zoweVersion | tee -a $LOG_FILE
+else
+    echo Error: Unable to obtain zoweVersion from $runtimePath/manifest.json | tee -a $LOG_FILE
+    exit 1
+fi
+
+# Verify hash file directory contents minimally
+if [[ ! -f $hashPath/HashFiles.class ]]
+then
+    echo Error: hashPath $hashPath does not contain HashFiles.class file | tee -a $LOG_FILE
+    exit 1
+fi 
+
+RefRuntimeHash=RefRuntimeHash-$zoweVersion.txt
+# Verify refPath directory contents minimally
+if [[ ! -f $refPath/$RefRuntimeHash ]]
+then
+    echo Error: refPath $refPath does not contain $RefRuntimeHash | tee -a $LOG_FILE
+    exit 1
+fi
+
+# Verify refPath contents minimally
+head $refPath/$RefRuntimeHash|grep '^\./' 1> /dev/null 2>/dev/null
+if [[ $? -ne 0 ]]
+then
+    echo Error: Lines at top of file $refPath/$RefRuntimeHash do not start with \"./\" | tee -a $LOG_FILE
+    exit 1
+fi 
+
+echo Info: Gathering files ... | tee -a $LOG_FILE
+
+cd $runtimePath
+. bin/internal/zowe-set-env.sh # ensure we have tagging behaviour set correctly
```

Comment:
I think this should go at the top of the script like in the others so that the environment is setup properly from the start

Reply:
I could move it up, but `runtimePath`  isn't verified until line 195.  Lines above that would still be vulnerable.